Fix #3092: enforce JavaScript and NodeJS spelling

DeekshaVarshney123 opened the pull request:
Hii @stevepiercy ,

I’ve attempted to fix issue #3092 by adding JavaScript and NodeJS to the Plone Vale accept vocabulary to ensure consistent spelling in the documentation.

Please review and let me know if any changes are required.
Thank you!

stevepiercy reviewed (changes requested):
All of the 17 items in `packages/**/news/` should not appear in this pull request. Only the one that you need to create for this PR should appear. I'm not sure what you did with git or how to resolve this part.

I noticed also that developers misnamed Node.js as either "node" or "Node" in numerous places in the docs, especially the change log. To fix that, please add `(?i)node` to the `reject.txt` file and replace them as needed. Note that it's possible that the word "node" is used per the English dictionary definition, in which case those instances should remain as "node". Keep in mind we want to correct the spelling of the product "Node.js" only.

DeekshaVarshney123 committed:
Fix #3092: Correct Node.js spelling and update vocab rules

DeekshaVarshney123 commented on docs/source/release-notes/index.md:
```diff
@@ -3259,7 +3259,7 @@ myst:
 ### Documentation
 
-- Added `JavaScript` and `NodeJS` as accepted spellings, and deviations of them as rejected spellings. @utkkkarshhh [#3092](https://github.com/plone/volto/issues/3092)
+- Added `JavaScript` and `Node.js` as accepted spellings, and deviations of them as rejected spellings. @utkkkarshhh [#3092](https://github.com/plone/volto/issues/3092)
```
okk

DeekshaVarshney123 commented:
Hi @stevepiercy,

I've updated the PR. I accidentally included the correct spelling in the reject.txt earlier, but I've fixed it now to only reject NodeJS and (?i)node. I also rebased the branch to clear out the extra news items and updated 'Node' to 'Node.js' where it refers to the product.

Regarding the failing Acceptance Test (24.x), could you please check if it's related to my changes or if it's a flaky test? 
Thank you!

stevepiercy commented on docs/source/upgrade-guide/index.md:
```diff
@@ -2898,7 +2898,7 @@ This release includes a number of changes to the internal dependencies. If you h
 ### Upgrade to Node.js 12
 
-We have now dependencies that requires `node >=10.19.0`. Although Node.js 10 has still LTS
+We have now dependencies that requires `Node.js >=10.19.0`. Although Node.js 10 has still LTS
```
This one should be reverted because that's how we declare a dependency in code.

stevepiercy reviewed (changes requested):
Almost all of the changes look good, except for a couple. Please see my comments for details. Thank you!

DeekshaVarshney123 committed:
Final fixes: escaped Node.js in vocab, reverted code strings, and added credit

stevepiercy commented on packages/volto/news/3092.internal:
I just noticed the extension of this file. Please rename it to `3092.documentation`.

stevepiercy commented:
@DeekshaVarshney123 I'm sorry, but I can't spare time to give you explicit instructions for how to resolve the issues. At that point, I'd be doing the work myself, and you wouldn't get credit.

Configuring Vale is tricky, and requires repeated testing until the regular expressions are exactly right. It's especially tricky because we don't want false positives or negatives in the results. I hope that my first attempt to resolve the issues leads you to discover a better resolution.

The best direction I can give you is the following.

- Accept only `Node.js` as the official spelling.
- Reject any short form or incorrect spelling or capitalization. These include, but are not limited to:
  - `node`
  - `Node`
  - `NodeJS`
  - `Node.JS`
- Refer to https://6.docs.plone.org/contributing/documentation/authors.html#advanced-vale-usage and https://vale.sh/docs. I'd suggest creating a sample file to play around with that has all spelling variations of Node.js, and using the single file command to speed up testing.

DeekshaVarshney123 committed:
Merge branch 'main' into spellcheck-fix
